### ISSUE TITLE
An app with the SDK integrated fails to load MapboxCoreMaps via dyld when run on a physical device.

### ISSUE THREAD

**Issue by klundberg** [mapbox/mapbox-maps-ios#670]
## Environment

- Xcode version: 12.5.1
- iOS version: 14.7.1
- Devices affected: iPhone 12, maybe more
- Maps SDK Version: 10.0.0 rc8

## Observed behavior and steps to reproduce

Whenever I try to build an app that integrates the 10.0 sdk, I get a trap during dyld before the app can start, with the following error message about how MapboxCoreMaps can't be loaded, but only when building and running on a device:

```
dyld: Library not loaded: @rpath/MapboxCoreMaps.framework/MapboxCoreMaps
  Referenced from: <App path>/Frameworks/<Framework path>
  Reason: no suitable image found.  Did find:
	<App path>/Frameworks/<Framework path>/Frameworks/MapboxCoreMaps.framework/MapboxCoreMaps: code signature in (<App path>/Frameworks/<Framework path>/Frameworks/MapboxCoreMaps.framework/MapboxCoreMaps) not valid for use in process using Library Validation: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.
dyld: launch, loading dependent libraries
DYLD_LIBRARY_PATH=/usr/lib/system/introspection
DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
```

I currently use swift package manager to integrate mapbox maps, and link the MapboxMaps library into a dynamic framework target (referenced by the `<App path>/Frameworks/<Framework path>` redaction in the log text above.

Running on a simulator works just fine, and archiving a build and running the archived build works, only trying to build and debug directly to device fails with this error.

## Expected behavior

I'd expect the library to load properly on device when loading the app regardless of how I integrate it into my app (cocoapods vs spm, directly into my app vs in a shared framework vs a dependency of another swift package in my project)

## Notes / preliminary analysis

I do have a workaround that i've discovered. If I manually add the `MapboxCoreMaps` and `MapboxMobileEvents` swift packages to my project, and integrate them directly into the app target, this dyld error goes away, presumably because dyld isn't looking in the wrong place for those frameworks. I imagine that this isn't a problem if your app directly integrates `MapboxMaps` into its binary, but integrating it into a different target like a framework causes these embedding/linking problems.

This may just be a SPM issue but providing the above as documentation or an faq or something if it can't be fixed or worked around otherwise may be warranted.

**Comment by buechner** [comment]
I have the same problem with the current version 10.10.1.

**Comment by sobri909** [comment]
I've just had this problem migrating a 7 year old project to SwiftPM from CocoaPods. I've finally found the solution for my case, which is to add `@executable_path/Frameworks` to `Runpath Search Paths`. 

I found that in a new project that Build Setting was already "customised" to the above (as well as `$(inherited)` as first line), but in my 7 year old project there wan't any customised value. Adding it finally got past the problem of on-device execution failing to find the dyld. Phew!

I don't know how common this is. Perhaps there's not a lot of old projects out there migrating to SwiftPM? Though I suspect there are. It might help to have that `RunPath Search Paths` value mentioned in the Install docs.
